core/internal/testutils/evmtest/evmtest.go

```diff
@@ -36,7 +36,7 @@ import (
 )
 
 type TestChainOpts struct {
-	Client         evmclient.Client
+	Client         interface{} // evmclient.Client or func(evmtypes.DBChain) evmclient.Client
```

Comment:
I think we can use generics to keep things type safe:
```suggestion
	Client         interface{ evmclient.Client | func(evmtypes.DBChain) evmclient.Client }
```
Or maybe:
```go
type ClientOrFn interface{ evmclient.Client | func(evmtypes.DBChain) evmclient.Client }

type TestChainOpts struct {
    Client ClientOrFn
```

Reply:
It looks like it wasn't this simple. You might have to push the generic type up to the struct:
```go
type TestChainOpts[C interace{...}] struct {
    Client C
```
You should still be able to infer the type at the callsites fortunately, but it would spread to all the method receivers, so maybe not worth it for test code :shrug:

Reply:
Alright, decided to just roll it back.  I noticed you changed the signature for this function this morning also... got that merged in properly now.